### ISSUE TITLE
[Bug]: Smart Playlists - Genre 'does not contain' gives unexpected results with multiple genres

### ISSUE THREAD

**Issue by ScaldedWilder** [navidrome/navidrome#2708]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.50.2 (823bef54)

### Current Behavior

I have multi-genre tracks, and want to filter out one of the genres with a smart playlist. but I have seen many tracks (maybe all?) slipping through this filter. When investigating, tracks with the second genre 'LofiGirl' could be found in both playlists marked as 'does not contain' and 'does contain' 'LofiGirl' 

![image](https://github.com/navidrome/navidrome/assets/153934407/2920fab2-d34c-4d95-aef8-c6a2b20a20c5)

![image](https://github.com/navidrome/navidrome/assets/153934407/38a75bf6-9d8a-441c-95be-24cf89d5c32b)

![image](https://github.com/navidrome/navidrome/assets/153934407/2177a227-8509-497e-9b62-31445a0dd3cb)

### Expected Behavior

I expect that 'genre' 'does not contain' 'LofiGirl' would exclude any tracks with 'LofiGirl' as a genre. 

### Steps To Reproduce

1. Add file with multiple genres. In this case, second genre is LofiGirl
2. Add NSP file, something like:
```
{
  "all": [
    {"isNot": {"rating": 1}},
    {"isNot": {"rating": 2}},
    {"notContains": {"genre": "LofiGirl"}}
  ],
  "sort": "random",
  "limit": 300
}
```

3. The resulting smart filter fails to exclude the file containing the undesired tag 'LofiGirl'

### Environment

```markdown
- OS: Ubuntu 22.04.3 LTS (Running official docker image)
- Browser: librewolf, firefox
- Client: Feishin 0.5.3
```

### How Navidrome is installed?

Docker

### Configuration

```toml
services:
  navidrome:
    image: deluan/navidrome:latest
    user: 1001:1001 # should be owner of volumes - musicplayer
    ports:
      - "4533:4533"
    restart: unless-stopped
    environment:
      # Optional: put your config options customization here. Examples:
      ND_SCANSCHEDULE: 24h
      ND_LOGLEVEL: debug
      ND_SESSIONTIMEOUT: 24h
      ND_BASEURL: ""
      ND_ENABLECOVERANIMATION: "false"
      ND_SCANNER_GROUPALBUMRELEASES: "true"
    volumes:
      - "/srv/musicsrv/navidrome:/data"
      - "/mnt/zpool/Media/Music:/music:ro"
      - "/srv/musicsrv/navidrome/log:/var/log"
```

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by ScaldedWilder** [comment]
I've tested this more, to try and narrow this down more. 

It seems like it's possible 'does not contain' may only be getting applied correctly if there is only one genre.

A good test is using something like

* Contains: Alt
* Contains Rock
gives me entries with genres like:
* Alternative Rock
* Rock
* Pop Punk, Alternative rock, Rock

but
* Contains: Alt
* does not contain: Rock
yields songs with genres like:
* Alternative Metal, Rock, Alternative Rock
* Alternative Rock, Alternative Metal
* Alternative & Punk
* Alternative
But I don't see any _single_ genre songs with 'Rock' in the result

So I believe the reduced/core symptom is that 'Does not contain' appears to only correctly apply if there is a single genre. If there are multiple genres, even matching the first genre (e.g. Rock, Alternative) seems insufficient for smart playlists to omit the song.

**Comment by metalheim** [comment]
Can confirm this behaviour. 
"genre isnot X" or "genre doesnotcontain X" only filters out tracks that have this genre X *and only X*
As soon as a track has multiple genres, this filter kinda becomes useless. 

My guess is that this filter just translates to filtering the `media_file_genres` table and then grouping them by song again.

**Comment by mneuburg-livefront** [comment]
But I've seen some other genre-related weirdness related to the 0.50.x versions. I had to go back to 0.49.3 to work around this. Something changed here.

**Comment by ScaldedWilder** [comment]
The opening issue (LofiGirl genre not being correctly filtered out) existed in 0.49.3, but I just lived with it, figuring I'd done something wrong. After I upgraded and it persisted, I finally decided to debug some and file this issue.

I'd recommend filing separate issues for the genre bugs you've experienced that's new in 0.50.X

**Comment by mneuburg-livefront** [comment]
Agreed and apologies. My issue turned out to be https://github.com/navidrome/navidrome/issues/2728.

**Comment by jlcala** [comment]
I have 3+ genres per track and can confirm this behavior as well. Navidrome has correctly parsed the genres (they are stored individually when I check Get Info), but does not seem to use all of the genres for each track to determine the smart playlist matches.

**Comment by mikaelmantel** [comment]
I have a lot of songs with mutiples genres, and the "contains" filter does not work properly with my songs.
How do you bypass this problem ?

Thank you

**Comment by ejalal** [comment]
Similar issue here where I still get songs with genres Electronic ect... when I only want Rock songs, with this smart playlist

```
{
  "all": [
    {
      "gt": {
        "rating": 3
      }
    },
    {
      "is": {
        "genre": "Rock"
      }
    },
    {
      "all": [
        {
          "isNot": {
            "genre": "Electronic"
          }
        },
        {
          "isNot": {
            "genre": "Reggae"
          }
        },
        {
          "isNot": {
            "genre": "Pop"
          }
        }
      ]
    }
  ]
}
```

**Comment by deluan** [comment]
Yeah, this is a known issue, but handling multi-valued tags (genres in this case) in Smart Playlists is completely being redone as part of #2709, so unless someone contributes with a simple PR to quickly fix this, you'll have to wait for #2709 to be complete

**Comment by Szeraax** [comment]
As this is not a critical bug disrupting most ND users, I'm removing triage.
